nomenclature/codelist.py

```diff
@@ -145,6 +145,17 @@ def check_stray_tag(cls, v):
                 )
         return v
 
+    @validator("mapping")
+    def check_end_whitespace(cls, v):
+        """Check that no code ends with a whitespace"""
+        print(v)
+        for code in v:
+            if code.endswith(" "):
+                raise ValueError(
+                    f"Unexpected white space at the end of a code: {code}."
```

Comment:
Since this error is specifically about white space it might not be a bad idea to put the `{code}` value in quotes:
```suggestion
                    f"Unexpected white space at the end of a code: '{code}'."
```
This way it should be more obvious to the user that there is in fact a white space character at the end of the code.